app/src/main/java/org/onionshare/android/ui/settings/SettingsTorUi.kt

```diff
@@ -0,0 +1,287 @@
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.OutlinedTextField
+import androidx.compose.material.RadioButton
+import androidx.compose.material.Scaffold
+import androidx.compose.material.SnackbarDuration
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.material.TextButton
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ContentPaste
+import androidx.compose.material.rememberScaffoldState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment.Companion.CenterVertically
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.platform.LocalClipboardManager
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextRange
+import androidx.compose.ui.text.font.FontFamily
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.navigation.NavHostController
+import kotlinx.coroutines.launch
+import org.onionshare.android.R
+import org.onionshare.android.ui.AboutActionBar
+import org.onionshare.android.ui.MainViewModel
+import org.onionshare.android.ui.ROUTE_SETTINGS_MY_BRIDGES
+import org.onionshare.android.ui.ShareButton
+import org.onionshare.android.ui.theme.OnionBlue
+import org.onionshare.android.ui.theme.OnionshareTheme
+import kotlin.random.Random
+
+private const val BRIDGE_DB = "https://bridges.torproject.org"
+
+@Composable
+fun SettingsTorUi(
+    navController: NavHostController,
+    viewModel: MainViewModel = hiltViewModel(),
+) {
+    val context = LocalContext.current
+    val settingsManager = viewModel.settingsManager
+    val scaffoldState = rememberScaffoldState()
+    val coroutineScope = rememberCoroutineScope()
+    Scaffold(
+        scaffoldState = scaffoldState,
+        topBar = {
+            AboutActionBar(navController, R.string.settings_tor_title)
+        }
+    ) { innerPadding ->
+        SettingsTorUiContent(
+            country = settingsManager.currentCountry,
+            automaticBridges = settingsManager.automaticBridges.value,
+            numberOfCustomBridges = settingsManager.customBridges.value.size,
```

Comment:
With LiveData there's a risk when calling getValue() that the value may not have been initialised yet (if it wasn't initialised in the constructor). Does the same risk exist for State, or is it guaranteed that the value will have been loaded from shared preferences before getValue() returns?

Reply:
I agree, better to do it on the UI thread. Thanks for explaining!